internal/action/edit.go

```diff
@@ -46,26 +46,32 @@ func (s *Action) edit(ctx context.Context, c *cli.Context, name string) error {
 		content = bytes.TrimPrefix(content, []byte(secret.Ident+"\n"))
 	}
 
+	// preserve intermediate state across retries
+	tmpContent := content
 	for {
 		// invoke the editor to let the user edit the content
-		nContent, err := editor.Invoke(ctx, ed, content)
+		newContent, err := editor.Invoke(ctx, ed, tmpContent)
 		if err != nil {
 			return ExitError(ExitUnknown, err, "failed to invoke editor: %s", err)
 		}
+		tmpContent = newContent
 		if fromMIME {
```

Comment:
What happens if it is not `fromMime` ?? Is it then not parsed at all?
We might want to do the same thing for insert wrt. #1602

Reply:
Yes, if it's not fromMime it's not parsed at all.
And yes, we might want to do that for insert as well. But certainly in a different PR.

Reply:
Agreed